app/(frontend)/auth/login/page.tsx

```diff
@@ -44,6 +66,9 @@ export default function LoginPage() {
                         variant="underlined"
                         className="w-3/4"
                         isRequired
+
+                        isInvalid={typeof state !== "string" && state?.email !== undefined}
+                        errorMessage={typeof state !== "string" && state?.email}
```

Comment:
**Refactor into function**
Also looks awfully similar to the logic in Line 70. Could you maybe find a solution which covers both of these cases?